File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null){
            return null;
        }
        ListNode cur = head;
        int tot = 0;
        while (cur!=null){
            tot += 1;
            cur = cur.next;
        }
        if(tot == 1 && n==1){
            return null;
        }
        int curPos = tot-n;
        if (curPos == 0) {
            return head.next;
        }
        ListNode cur1 = head;
        for (int i=0;i<curPos-1; i++){
            cur1 = cur1.next;
        }
        cur1.next = cur1.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        curr = head
        tot = 0
        while curr:
            tot += 1
            curr = curr.next
        if n == 1 and tot == 1:
            return None
        currPos = tot - n
        if currPos == 0:
            return head.next
        curr1 = head
        for i in range(currPos-1):
            curr1 = curr1.next
        curr1.next = curr1.next.next

        return head
